### PR TITLE
Adding UI theam for app type and application listing

### PR DESCRIPTION
**Pull request by lmanohara**
- Application listing page template
- Application type page template

### CODE REVIEW

File: modules/jaggeryapps/appmgt/src/jagg/constants.jag [hunk 1]
```diff
@@ -23,5 +23,6 @@ var SUPER_TENANT_DOMAIN_NAME = "carbon.super";
 var CLOUD_TYPE="AppCloud";
 var DOCUMENTATION_REDIRECTION_URL = "https://docs.wso2.org/display/AF210/WSO2+App+Factory+Documentation";
 var SUPPORT_REDIRECTION_URL = "http://wso2.com/support/";
+var APP_TYPES_PROPERTIES_CONFIG = "/site/themes/default/templates/apptype/select/app-types-properties.json";
```
Comment (danushkaf) [inline modules/jaggeryapps/appmgt/src/jagg/constants.jag]: Shouldn't we need to consider the file paths for other systems. We need to use File.Seperator IMO

File: modules/jaggeryapps/appmgt/src/jagg/constants.jag [hunk 2]
```diff
@@ -23,5 +23,6 @@ var SUPER_TENANT_DOMAIN_NAME = "carbon.super";
 var CLOUD_TYPE="AppCloud";
```
Comment (punnadi) [inline modules/jaggeryapps/appmgt/src/jagg/constants.jag]: Place spaces between = as given below:

var CLOUD_TYPE = "AppCloud";

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag [hunk 3]
```diff
@@ -23,65 +23,146 @@ jagg.template("apptype/select", function(inputs, outputs, jagg) {
...
-                                <br>
-                                </div>
-                                <%
-                            }
-                        }
-                    %>
-                    <!--<h2><a href="/appmgt/site/pages/select-apptype.jag">Create</a> a new application</h2>-->
+    <div class="container-fluid app-type-listing">
+       <div class="row listing padding-bottom-xlg" data-count="3.0">
+          <%
```
Comment (punnadi) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: Check the formatting

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag [hunk 4]
```diff
@@ -23,65 +23,146 @@ jagg.template("apptype/select", function(inputs, outputs, jagg) {
...
+            $(this).addClass('cloud-app-selected');
+        }else{
+            $(this).addClass('cloud-app-selected');
+
+        }
+
+        var width = $( window ).width(),
+                currentcount = parseInt($(this).attr('id')),
+                appDescription = $(this).attr('data-description'),
+                appName = $(this).attr('data-appname'),
```
Comment (punnadi) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: Check the formatting

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag [hunk 5]
```diff
@@ -23,65 +23,146 @@ jagg.template("apptype/select", function(inputs, outputs, jagg) {
...
+                $('.longme').fadeIn('slow')
+            }else{
+                var ctest = parseInt(currentcount)+(4- currentcount%4);
+                if(ctest > dataCount){
+                    $('#'+ dataCount+'\\.0').parent().after(appendHtml);
+                    $('.longme').fadeIn('slow')
+                }else{
+                    $('#'+ctest+'\\.0').parent().after(appendHtml);
+                    $('.longme').fadeIn('slow')
+
```
Comment (punnadi) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: remove space

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag [hunk 6]
```diff
@@ -23,65 +23,146 @@ jagg.template("apptype/select", function(inputs, outputs, jagg) {
...
+        var appCreationPage = '<%=jagg.getAbsoluteUrl("/site/pages/application.jag")%>?appTypeName='+ appType;
+
+        //content replace with data attributes
+        $('.app-type-info-template').find('.data-appname').html(appName);
+        $('.app-type-info-template').find('.app-description').html(appDescription);
+        $('.app-type-info-template').find("a").attr("href", appCreationPage);
+        var appendHtml =$('.app-type-info-template').html();
+
+
+        if(width >=1170){
```
Comment (punnadi) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: Comment on these numbers for more clarity (1170 etc)

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag [hunk 7]
```diff
@@ -23,65 +23,146 @@ jagg.template("apptype/select", function(inputs, outputs, jagg) {
...
+                $('.longme').fadeIn('slow')
+            }else{
+                var ctest = parseInt(currentcount)+1;
+                if(ctest > dataCount){
+                    $('#'+ dataCount+'\\.0').parent().after(appendHtml);
+                    $('.longme').fadeIn('slow')
+                }else{
+                    $('#'+ctest+'\\.0').parent().after(appendHtml);
+                    $('.longme').fadeIn('slow')
+
```
Comment (punnadi) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: Spaces

File: modules/jaggeryapps/appmgt/src/jagg/constants.jag (hunk 1 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/jagg/constants.jag]: This is fixed in following commit.
https://github.com/wso2/app-cloud/pull/167/commits/7cd8e546e662d9526c2c7714709e434fd00c7ff7

File: modules/jaggeryapps/appmgt/src/jagg/constants.jag (hunk 2 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/jagg/constants.jag]: This is fixed in following commit.
https://github.com/wso2/app-cloud/pull/167/commits/7cd8e546e662d9526c2c7714709e434fd00c7ff7

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag (hunk 3 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: This is fixed in following commit.
https://github.com/wso2/app-cloud/pull/167/commits/7cd8e546e662d9526c2c7714709e434fd00c7ff7

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag (hunk 4 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: This is fixed in following commit.
https://github.com/wso2/app-cloud/pull/167/commits/7cd8e546e662d9526c2c7714709e434fd00c7ff7

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag (hunk 5 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: This is fixed in following commit.
https://github.com/wso2/app-cloud/pull/167/commits/7cd8e546e662d9526c2c7714709e434fd00c7ff7

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag (hunk 6 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: This is fixed in following commit.
https://github.com/wso2/app-cloud/pull/167/commits/7cd8e546e662d9526c2c7714709e434fd00c7ff7

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag (hunk 7 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: This is fixed in following commit.
https://github.com/wso2/app-cloud/pull/167/commits/7cd8e546e662d9526c2c7714709e434fd00c7ff7

File: modules/jaggeryapps/appmgt/src/site/blocks/apptype/select/block.jag [hunk 8]
```diff
@@ -30,6 +32,13 @@ jagg.block("apptype/select", {
         var applicationModule = jagg.module("application");
         var appTypes = applicationModule.getAllAppTypes();
 
+        var appTypesProperties = require(APP_TYPES_PROPERTIES_CONFIG);
+        if(appTypes.length > 0) {
```
Comment (prabathabey) [inline modules/jaggeryapps/appmgt/src/site/blocks/apptype/select/block.jag]: There are code formatting issues in this PR as well. Have these already been fixed?

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag [hunk 9]
```diff
@@ -23,65 +23,146 @@ jagg.template("apptype/select", function(inputs, outputs, jagg) {
...
+                }
+
+            }
+
+        }else if(width <750 ){
+            $('#'+currentcount+'\\.0').parent().after(appendHtml);
+            $('.longme').fadeIn('slow')
+
+        }
+
```
Comment (imesh) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: Might be better to format the code and remove blank lines.

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag [hunk 10]
```diff
@@ -25,40 +25,46 @@ jagg.template("index", function(inputs, outputs, jagg) {
...
+                    </div>
+                    <div class="app-name" ><%=applications[x].applicationName%></div>
+                </div>
+             </a>
+           </div>
+        <%
+        }
+    } else { %>
+      <div>Please create a new application</div>
+    <%
```
Comment (imesh) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: Please fix indentation.

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag (hunk 9 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: This is fixed in following pull request.
https://github.com/wso2/app-cloud/pull/205/files#diff-f64267f1c0cc29adaf918bd5fcf877e2R37

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag (hunk 10 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/index/template.jag]: This is fixed in following pull request.
https://github.com/wso2/app-cloud/pull/205/files#diff-f64267f1c0cc29adaf918bd5fcf877e2R37

File: modules/jaggeryapps/appmgt/src/site/blocks/apptype/select/block.jag (hunk 8 above)
Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/blocks/apptype/select/block.jag]: This is fixed in following pull request.
https://github.com/wso2/app-cloud/pull/262

File: modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag [hunk 11]
```diff
@@ -23,65 +23,146 @@ jagg.template("apptype/select", function(inputs, outputs, jagg) {
...
+        var appendHtml =$('.app-type-info-template').html();
+
+
+        if(width >=1170){
+            if(currentcount%7 == 0){
+                $('#'+currentcount+'\\.0').parent().after(appendHtml);
+                $('.longme').fadeIn('slow')
+            }else{
+                var ctest = currentcount+(7- currentcount%7);
+                console.log(ctest)
```
Comment (lalaji) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: Is this console.log is needed?

Comment (lmanohara) [inline modules/jaggeryapps/appmgt/src/site/themes/default/templates/apptype/select/template.jag]: No. This is fixed in following pull request.
https://github.com/wso2/app-cloud/pull/205/commits/7cd8e546e662d9526c2c7714709e434fd00c7ff7

**Pull request merged**